modules/core/shared/src/main/scala/io/circe/Json.scala

```diff
@@ -584,13 +584,14 @@ object Json {
    */
   private[this] def isReal(value: Float): Boolean = java.lang.Float.isFinite(value)
 
-  private[this] final def arrayEq(x: Seq[Json], y: Seq[Json]): Boolean = {
+  private[this] final def arrayEq(x: Vector[Json], y: Vector[Json]): Boolean = {
+    if (x.length != y.length) return false
     val it0 = x.iterator
     val it1 = y.iterator
-    while (it0.hasNext && it1.hasNext) {
+    while (it0.hasNext) {
       if (Json.eqJson.neqv(it0.next(), it1.next())) return false
     }
-    it0.hasNext == it1.hasNext
+    true
   }
```

Comment:
I wonder if we could just call into the standard library equality check for vectors and make this a shorter/faster implementation?

Reply:
Interesting point, I think it would definitely make the code shorter not sure about squeezing extra performance.

At the same time, I do not have the context regarding why it was done like this in first place.
The spirit of the PR is to shortcut the existing implementation, not to modify it.
 
Pushing for a bigger change would require a JMH benchmark IMHO.